forge/csrc/passes/extract_unique_op_configuration.cpp

```diff
@@ -0,0 +1,228 @@
+            operand_shapes.push_back(operand->shape());
+        }
+
+        // If the op is present in the unique_op_shapes and unique_op_attrs unordered map,
+        // then list of operand shapes of the matched op is compared with the current node operand shapes
+        // otherwise the current node operand_shapes and attributes(i.e OpTypes) are added to the
+        // unique_op_shapes and unique_op_attrs with new unique_id.
+        if ((unique_op_shapes.find(current_node_optype.op) != unique_op_shapes.end()) &&
+            (unique_op_attrs.find(current_node_optype.op) != unique_op_attrs.end()))
+        {
+            auto unique_op_id_and_opshapes = unique_op_shapes.at(current_node_optype.op);
+            auto unique_op_id_and_optypes = unique_op_attrs.at(current_node_optype.op);
+            bool operand_shapes_matched = false;
+            for (auto iter = unique_op_id_and_opshapes.begin(); iter != unique_op_id_and_opshapes.end(); ++iter)
+            {
+                auto unique_op_id = iter->first;
+                auto unique_op_shapes = iter->second;
+                // If the current node and matched op operand shapes are equivalent,
+                // then take the list of OpType from the unique_op_attrs unordered map with matched op name and
+                // unique_id from the matched operand shapes and then compare with  current node optype, if the current
+                // node optype is not present, then add current node optype in unique_op_attrs with matched op name and
+                // unique_id from matched operand shapes.
+                if (equivalent_shapes(unique_op_shapes, operand_shapes) &&
+                    (unique_op_id_and_optypes.find(unique_op_id) != unique_op_id_and_optypes.end()))
+                {
+                    operand_shapes_matched = true;
+                    auto unique_optypes = unique_op_id_and_optypes.at(unique_op_id);
+                    if (std::find(unique_optypes.begin(), unique_optypes.end(), current_node_optype) ==
+                        unique_optypes.end())
+                    {
+                        unique_op_attrs[current_node_optype.op][unique_op_id].push_back(current_node_optype);
+                    }
+                }
+            }
+            if (!operand_shapes_matched)
+            {
+                unique_op_shapes[current_node_optype.op][unique_id] = operand_shapes;
+                unique_op_attrs[current_node_optype.op][unique_id] = {current_node_optype};
+                unique_id++;
+            }
+        }
+        else
+        {
+            unique_op_shapes[current_node_optype.op] = {{unique_id, operand_shapes}};
+            unique_op_attrs[current_node_optype.op] = {{unique_id, {current_node_optype}}};
+            unique_id++;
+        }
+    }
+
+    if (unique_op_shapes.empty() and unique_op_attrs.empty())
+        return;
+
+    // print the unique ops configurations that are present in the gragh
+    if (print_unique_op_config)
```

Comment:
Let's think about splitting print and export logic into separate functions

Reply:
I will make it as separate function